### PR TITLE
Istio, automount serviceAccountToken when Istio enabled in VMI annotations

### PR DESCRIPTION
**Pull request by rhrazdil**
**What this PR does / why we need it**:

This PR adds default serviceAccountToken automatically when
sidecar.istio.io/inject annotation is added in the VMI.

This improves user experience, bacause users don't need to add a serviceAccountDisk manually.
Adding the annotation becomes the only requirement for Istio service mesh to work properly.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
Add serviceAccountDisk automatically when Istio is enabled in VMI annotations
```

### CODE REVIEW

**Comment by xpivarc** [comment]
/assign
I would like to review this before merging. I am actually working on feature that uses SA and it could collide  with this feature.

**Comment by rhrazdil** [comment]
@xpivarc  Many thanks!

File: pkg/virt-controller/services/template.go
Comment (xpivarc) [inline pkg/virt-controller/services/template.go:461]: Is it sufficient for Istio to inject any service account?
We can maybe do something like this:
```

automount := true
pod.Spec.AutomountServiceAccountToken = &automount
```
if for any reason the default account is different.

Comment (rhrazdil) [reply pkg/virt-controller/services/template.go]: >Is it sufficient for Istio to inject any service account?

To my understanding, yes. [there is a requirement for NET_RAW and NET_ADMIN](https://istio.io/latest/docs/ops/deployment/requirements/), but we intent to only use Istio with CNI Plugin. 
They don't mention any other requirement for the service account.

So I believe you have a point. [Updated the accordingly to your suggestion](https://github.com/kubevirt/kubevirt/compare/eb0cee1722d0ca2845285dc7c66548fdaa75264d..262e988b3b60906d77210e48974869b290dc2e22)

Comment (rhrazdil) [reply pkg/virt-controller/services/template.go]: Right, thanks. [Done](https://github.com/kubevirt/kubevirt/compare/262e988b3b60906d77210e48974869b290dc2e22..73ac0b974aec61580368030c887b39c44f350241), also removed the now unused constant.

**Review by xpivarc — APPROVED** [review]

File: pkg/virt-controller/services/template.go
Comment (maiqueb) [inline pkg/virt-controller/services/template.go:90]: I don't like the fact that we're duplicating this from https://github.com/kubevirt/kubevirt/blob/37ec453d972a7b4957fb6c1160cfdba71d007512/pkg/virt-launcher/virtwrap/network/podinterface.go#L60.

Would you consider defining this in `.../pkg/network/...` or a network types / constants package under that tree ? 

The PR can go in as is, I'd just like to establish some ground rules on how to deal w/ this for the future.

File: pkg/virt-controller/services/template.go
Comment (maiqueb) [inline pkg/virt-controller/services/template.go:1320]: it reads well enough, but possibly exporting it to a tiny func - `hasIstioAnnotation` maybe - would make it better. Not a must - I think it reads well enough.

File: pkg/virt-controller/services/template_test.go
Comment (maiqueb) [inline pkg/virt-controller/services/template_test.go:897]: this is me being extremely picky, you can disregard this comment; it would never the allow the reader to be more focused if you'd create the `vmi` and `pod` objects in the test setup stage - `BeforeEach` - under this context.

**Review by maiqueb — COMMENTED** [review]
The PR is currently fine, but just some nits I've spotted.

File: pkg/virt-controller/services/template.go
Comment (rhrazdil) [reply pkg/virt-controller/services/template.go]: Done in the [1. commit ](https://github.com/kubevirt/kubevirt/pull/5797/commits/356690e9eb7d7c55f6593783358c991d9d7f95f2)
Added [2. commit](https://github.com/kubevirt/kubevirt/pull/5797/commits/588c3d10bb4eb751ab93824cf0f9a0293b1acdf4) where I use the defined constant in other places

File: pkg/virt-controller/services/template_test.go
Comment (rhrazdil) [reply pkg/virt-controller/services/template_test.go]: Yep. I like your idea. Done, thanks!

**Review by maiqueb — APPROVED** [review]
Thanks.

/hold cancel

**Pull request merged**
